memory-game.js

```diff
@@ -1,22 +1,146 @@
+// an id, a name and a picture url
+const imagesWeather = [
+  {
+    name: "sunny",
+    url: "./assets/sunny.png",
+    "back-cover": "./assets/back.jpg",
+    id: 0,
+    isFlipped: false,
+    isMatched: false,
+    value: "sol",
+  },
+  {
+    name: "cloudy",
+    url: "./assets/cloudy.png",
+    "back-cover": "./assets/back.jpg",
+    id: 1,
+    isFlipped: false,
+    isMatched: false,
+    value: "overskyet",
+  },
+  {
+    name: "rainy",
+    url: "./assets/rainy.png",
+    "back-cover": "./assets/back.jpg",
+    id: 2,
+    isFlipped: false,
+    isMatched: false,
+    value: "regn",
+  },
+  {
+    name: "snowy",
+    url: "./assets/snowy.png",
+    "back-cover": "./assets/back.jpg",
+    id: 3,
+    isFlipped: false,
+    isMatched: false,
+    value: "sne",
+  },
+];
+const images = [
+  {
+    name: "fox",
+    url: "./assets/fox.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 0,
+    isFlipped: false,
+    isMatched: false,
+    value: "ræv",
+  },
+
+  {
+    name: "dog",
+    url: "./assets/dog.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 2,
+    isFlipped: false,
+    isMatched: false,
+    value: "hund",
+  },
+  {
+    name: "cat",
+    url: "./assets/cat.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 3,
+    isFlipped: false,
+    isMatched: false,
+    value: "kat",
+  },
+  {
+    name: "horse",
+    url: "./assets/horse.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 4,
+    isFlipped: false,
+    isMatched: false,
+    value: "heste",
+  },
+  {
+    name: "bear",
+    url: "./assets/bear.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 5,
+    isFlipped: false,
+    isMatched: false,
+    value: "bjørn",
+  },
+  {
+    name: "eagle",
+    url: "./assets/eagle.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 6,
+    isFlipped: false,
+    isMatched: false,
+    value: "ørn",
+  },
+  {
+    name: "giraffe",
+    url: "./assets/giraffe.jpg",
+    "back-cover": "./assets/back.jpg",
+    id: 7,
+    isFlipped: false,
+    isMatched: false,
+    value: "giraf",
+  },
+];
+const animals = ["ræv", "hund", "kat", "hest", "bjørn", "ørn", "giraf"];
+const weather = ["sol", "overskyet", "regn", "sne"];
+//get the grid container
+//create a card with a front and back image
+//add the card to the grid container
+
 document.addEventListener("DOMContentLoaded", function () {
   const grid = document.getElementById("grid-container");
-  function createCard(imgFrontSrc, imgBackSrc) {
+
+  function shuffle(array) {
+    array.sort(() => Math.random() - 0.5);
+    return array;
+  }
```

Comment:
The eventListener `DOMContentLoaded` is starting to become a rather large function. Consider moving out your functions 

```js
document.addEventListener("DOMContentLoaded", function () {
  const grid = document.getElementById("grid-container");

  document.getElementById("animals-btn").addEventListener("click", () => {
    populateGrid("animals");
  });
  document.getElementById("weather-btn").addEventListener("click", () => {
    populateGrid("weather");
  });
  populateGrid("animals");
});

function shuffle(array) {
  ...
}

function createCard(type, data, imgBackSrc) {
  ...
}

function populateGrid(category) {
  ...
}
```

Reply:
yes, that makes sense!